### PR TITLE
Upgrade client-go and related to 1.15

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

This upgrade client-go, apimachinery, apiextensions-apiserver and related to support Kubernetes 1.15, as well as regenerated clients for our CRDs.

Also included is a fix to gen-crd-client.sh such that it can now use Go modules, as otherwise it was failing to generate clients.

**Which issue(s) this PR fixes**:

Work on #1478

**Special notes for your reviewer**:

The only edited files that are probably worth looking at are Dockerfile, gen-crd-client.sh and go.mod.
Everything else is generated / vendoring.

### CODE REVIEW

**Comment by markmandel** [comment]
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19faac0]

goroutine 1218 [running]:
testing.tRunner.func1(0xc0005d2100)
	/usr/local/go/src/testing/testing.go:874 +0x69f
panic(0x1bc19e0, 0x2b21820)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
agones.dev/agones/pkg/gameservers.(*Controller).createGameServerPod(0xc0003025a0, 0xc001187b00, 0x1d61b72, 0x8, 0xc000806000)
	/go/src/agones.dev/agones/pkg/gameservers/controller.go:596 +0x1220
agones.dev/agones/pkg/gameservers.(*Controller).syncGameServerCreatingState(0xc0003025a0, 0xc001187b00, 0xc0002bd130, 0xc0000ef080, 0xc0000eefd0)
	/go/src/agones.dev/agones/pkg/gameservers/controller.go:494 +0x5e9
agones.dev/agones/pkg/gameservers.TestControllerSyncGameServerCreatingState.func3(0xc0005d2100)
	/go/src/agones.dev/agones/pkg/gameservers/controller_test.go:675 +0x980
testing.tRunner(0xc0005d2100, 0xc0003d4fd0)
	/usr/local/go/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x652
FAIL	agones.dev/agones/pkg/gameservers	5.381s
ok  	agones.dev/agones/pkg/gameserversets	3.218s
```

Putting this here so it's easier to lookup, and find out what the culprit is when I dig into this.

File: go.mod [hunk 1]
```diff
@@ -45,11 +43,10 @@ require (
 	gopkg.in/fsnotify.v1 v1.4.7
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0-20170531160350-a96e63847dc3
-	k8s.io/api v0.0.0-20191004102349-159aefb8556b // kubernetes-1.14.10
-	k8s.io/apiextensions-apiserver v0.0.0-20191212015246-8fe0c124fb40 // kubernetes-1.14.10
-	k8s.io/apimachinery v0.0.0-20191004074956-c5d2f014d689 // kubernetes-1.14.10
-	k8s.io/client-go v11.0.1-0.20191029005444-8e4128053008+incompatible // kubernetes-1.14.10
+	k8s.io/api v0.15.11 // kubernetes-1.14.10
```
Comment (roberthbailey) [inline go.mod:46]: I'm not sure where these comments came from (did we add them by hand) but they are now out of date.

Comment (markmandel) [reply go.mod:46]: go.mod is generated by the go tooling. I've bo idea where they came from.

I'll edit them by hand.

Comment (markmandel) [reply go.mod]: Done!

**Pull request merged**